src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests3.cs

```diff
@@ -6415,10 +6490,56 @@ .maxstack  1
 }");
         }
 
+        [Theory]
+        [InlineData("c is not { F: var f } c1")]
+        public void IsNot_10b(string pattern)
+        {
+            var source =
+$@"using static System.Console;
+class C
+{{
+    static void Main()
+    {{
+        M(new C());
+    }}
+    static void M(C c)
+    {{
+        if ({pattern}) return;
+        WriteLine(f);
+        WriteLine(c1.F);
+    }}
+    object F = 42;
+}}";
+            var verifier = CompileAndVerify(source, expectedOutput:
+@"42
+42");
+            verifier.VerifyIL("C.M",
+@"{
+  // Code size       33 (0x21)
+  .maxstack  1
+  .locals init (C V_0, //c1
+                object V_1) //f
+  IL_0000:  ldarg.0
+  IL_0001:  stloc.0
+  IL_0002:  ldloc.0
+  IL_0003:  brfalse.s  IL_000e
+  IL_0005:  ldloc.0
+  IL_0006:  ldfld      ""object C.F""
+  IL_000b:  stloc.1
+  IL_000c:  br.s       IL_000f
+  IL_000e:  ret
+  IL_000f:  ldloc.1
+  IL_0010:  call       ""void System.Console.WriteLine(object)""
+  IL_0015:  ldloc.0
+  IL_0016:  ldfld      ""object C.F""
+  IL_001b:  call       ""void System.Console.WriteLine(object)""
+  IL_0020:  ret
+}");
+        }
+
         [Theory]
         [InlineData("!(o is (41 or 42))")]
-        [InlineData("o is not (41 or 42)")]
-        public void IsNot_11(string pattern)
+        public void IsNot_11a(string pattern)
```

Comment:
The reason for different codegen is that for some of tests we actually use linear lowering.

[Previously we did non-linear lowering for both cases](https://sharplab.io/#v2:D4AQTAjAsAUCDMACciDCiDetE8Qej0QGUB3ASwBcBjAC0QCMAnAQwDtaBTAZwY4pI4dWiACIcAZswCuAGwqI2AE0QBbZmWEUA9oi6DEFGh0SKy48R0ZCqxgHTZcCBlq0zEAFW4UIACi30AKw4qeS0ASkQAXgA+RB0yHlYteR8AFgg4xkRUsDCAbgccJ3oXN08uCjA/QODQiJjEAEI/RATsjK0snPzYAF8gA=).